### ISSUE TITLE
USB-C does not work

### ISSUE THREAD

**Issue by ThomasWaldmann** [QubesOS/qubes-issues#5522]
**Qubes OS version**
_The version of Qubes OS you're using (e.g., `R4.0`), available via the command `cat /etc/qubes-release` in a dom0 terminal._

R4.0 with all updates

**Affected component(s) or functionality**
_The component or functionality of Qubes OS that is not working as expected._

Kernel? dom0? sys-usb?

**Brief summary**

HW: Thinkpad T470s, Orico NVME M.2 SSD enclosure

If I plug in the USB-C SSD **into the USB-C/Thunderbolt port**, there is no reaction at all. lsusb in sys-usb does not show it, the usb menu does not show it.

If I plug it **into a normal USB3 port** (using a cable that converts from USB-C to USB-A connector), it works as expected: shows up in lsusb, shows up in the usb menu, can be assigned to a VM, can be used in a VM.

The external SSD case is based on the JMicron 152d:0583 USB-C to PCIe bridge and has a Samsung M.2 NVME/PCIe  SSD inside.

**Solutions you've tried**

I have installed all Qubes updates, did a cold boot of the machine.

Also checked the Thinkpad T470s' BIOS to see whether Thunderbolt/USB-C functionality is disabled (it is not, it is enabled).

Also cold booted with the USB-C device already attached, did not help.

Also tried another USB-C cable, did not help.

**Comment by DemiMarie** [comment]
> "no security" thunderbolt setting

This is a massive security risk, if I understand correctly.

The real problem is that dom0 does not support PCI hotplug.

**Comment by marmarek** [comment]
Yes, this is true. And unfortunately PCI hotplug + IOMMU situation in Xen is not great and for this reason we have it disabled.

So, the answer for now is "this feature is not supported", sadly.

**Comment by ThomasWaldmann** [comment]
Please note that I attach a USB-to-PCIe bridge.

So, as far as the laptop is concerned, this is a USB device, not a PCIe device.

**Comment by DemiMarie** [comment]
> Please note that I attach a USB-to-PCIe bridge.
> 
> So, as far as the laptop is concerned, this is a USB device, not a PCIe device.

On my laptop, plugging in a device into the USB-C port causes the USB controller to present a new PCI device.  So it appears as if a new USB PCI card was just ineserted.

**Comment by DemiMarie** [comment]
@marmarek are there any plans to fix this in Xen?

**Comment by ThomasWaldmann** [comment]
I diffed lspci output before vs. after plugging in the ssd.

No additional devices, the 5 devices have the same IDs as before, **but** their revision changes from `rev ff` to `rev 01`). Huh?

**Comment by ThomasWaldmann** [comment]
BTW (off-topic): are you at 36c3, @marmarek and @DemiMarie ?

**Comment by marmarek** [comment]
> I diffed lspci output before vs. after plugging in the ssd.
> No additional devices

In qubes - yes, of course, as we have PCI hotplug disabled.

> but their revision changes from rev ff to rev 01).

Where do you observe it? In dom0? or sys-usb?

> @marmarek are there any plans to fix this in Xen?

Unclear. There are people recognizing it as an important task, but without any specific timeline attached.

> BTW (off-topic): are you at 36c3, @marmarek and @DemiMarie ?

I'll be there.

**Comment by ThomasWaldmann** [comment]
I did lspci in dom0.

**Comment by ben-grande** [comment]
@DemiMarie On the other issue https://github.com/QubesOS/qubes-issues/issues/6620, you marked this one, #5522, as related but not duplicate. Can you clarify? This is Thunderbolt hotplug in dom0.
